price-feeder/config/config.go

```diff
@@ -186,12 +185,19 @@ func ParseConfig(configPath string) (Config, error) {
 		return cfg, ErrEmptyConfigPath
 	}
 
-	configData, err := os.ReadFile(configPath)
-	if err != nil {
-		return cfg, fmt.Errorf("failed to read config: %w", err)
+	viper.AutomaticEnv()
+	viper.SetConfigFile(configPath)
+	viper.SetConfigType("toml")
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			return cfg, fmt.Errorf("failed to read config because file was not found: %w", err)
+		} else {
+			return cfg, fmt.Errorf("failed to read config: %w", err)
+		}
```

Comment:
The underlying error will have the same information, might as well keep the code simple:
```suggestion
			return cfg, fmt.Errorf("failed to read config: %w", err)
```